src/services/kpis/kpis.svc.coffee

```diff
@@ -382,14 +382,33 @@ angular
         !alerts[alert.service].active
       )
 
+      # Update existing alerts that have been modified
+      alertsToUpdate = _.filter(alerts, (alert) ->
+        active_alert = _.filter(kpi.alerts, (kpi_alert) -> kpi_alert.service == alert.service)[0]
+        return false if alert.service == "inapp" || !active_alert
+        active_ids = active_alert.recipients.map((recipient) -> recipient.id).sort() if active_alert
+        recipient_ids = alert.recipient_ids.sort()
+        recipient_change = false
+        recipient_change = true if recipient_ids.length != active_ids.length
+        for i in [0...recipient_ids.length]
+          recipient_change = true if recipient_ids[i] != active_ids[i]
```

Comment:
Lets simplify the change detection above by comparing the arrays like so: `"#{[1, 3, 2].sort()}" == "#{[2, 1, 3].sort()}"`, also I think it should be clear with the variable naming what is existing and what is new. So naming `active_ids` to `existingActiveIds` etc.